common/flatpak-exports.c

```diff
@@ -860,27 +970,45 @@ _exports_path_expose (FlatpakExports *exports,
       if (slash)
         *slash = 0;
 
-      if (path_is_symlink (exports, path) && !never_export_as_symlink (path))
+      if (!path_is_symlink (exports, path))
+        {
+          flatpak_debug2 ("%s is not a symlink", path);
+        }
+      else if (never_export_as_symlink (path))
         {
-          g_autofree char *resolved = flatpak_exports_resolve_link_in_host (exports, path, NULL);
+          flatpak_debug2 ("%s is a symlink, but we avoid exporting it as such", path);
+        }
+      else
+        {
+          g_autoptr(GError) error = NULL;
+          g_autofree char *resolved = flatpak_exports_resolve_link_in_host (exports, path, &error);
           g_autofree char *new_target = NULL;
 
           if (resolved)
             {
+              flatpak_debug2 ("%s is a symlink, resolved to %s", path, resolved);
+
               if (slash)
                 new_target = g_build_filename (resolved, slash + 1, NULL);
               else
                 new_target = g_strdup (resolved);
 
+              flatpak_debug2 ("Trying to export the target instead: %s", new_target);
+
               if (_exports_path_expose (exports, mode, new_target, level + 1))
                 {
                   do_export_path (exports, path, FAKE_MODE_SYMLINK);
                   return TRUE;
                 }
             }
-
-          return FALSE;
+          else
+            {
+              flatpak_debug2 ("%s is a symlink but we were unable to resolve it: %s",
+                              path, error->message);
+              return FALSE;
+            }
```

Comment:
Hmm, looking at this again, putting this in `else` is a small behaviour change: if `_exports_path_expose()` returned false, previously we would return false here, but now we fall through.